raiden/storage/sqlite.py

```diff
@@ -351,6 +441,89 @@ def _query_events(self, limit: int = None, offset: int = None):
 
         return cursor.fetchall()
 
+    def _get_event_records(
+            self,
+            limit: int = None,
+            offset: int = None,
+            filters: Dict[str, Any] = None,
+    ) -> List[EventRecord]:
+        """ Return a batch of event records
+
+        The batch size can be tweaked with the `limit` and `offset` arguments.
+
+        Additionally the returned events can be optionally filtered with
+        the `filters` parameter to search for specific data in the event data.
+        """
+        limit, offset = _sanitize_limit_and_offset(limit, offset)
+        cursor = self.conn.cursor()
+
+        query = 'SELECT identifier, source_statechange_id, data FROM state_events '
+        where_clauses = []
+        args = []
+        if filters:
+            for field, value in filters.items():
+                where_clauses.append('json_extract(data, ?)=?')
+                args.append(f'$.{field}')
+                args.append(value)
+
+            query += (
+                f"WHERE {' AND '.join(where_clauses)}"
+            )
+
+        query += 'ORDER BY identifier ASC LIMIT ? OFFSET ?'
+        args.append(limit)
+        args.append(offset)
+
+        cursor.execute(query, args)
+        result = []
+        try:
+            rows = cursor.fetchall()
```

Comment:
here too we can simplify it to just a list comprehension

Reply:
Done in https://github.com/raiden-network/raiden/pull/3550/commits/73bb4205f6e611094beec393219463c0dd3dbb9c